### PR TITLE
Adding Cuda Variant to SW4Lite

### PR DESCRIPTION
**Pull request by rspavel**
Added cuda variant of sw4lite  as per guidance in SW4Lite README

### CODE REVIEW

File: var/spack/repos/builtin/packages/sw4lite/package.py [hunk 1]
```diff
@@ -52,6 +52,14 @@ def build_targets(self):
             cxxflags.append('-DSW4_CROUTINES')
             targets.append('ckernel=yes')
 
+        if '+cuda' in self.spec:
+            targets.append('NVCC = {0}'.format(
+                self.spec['cuda'].prefix.bin.nvcc))
+            targets.append('HOSTCOMP = {0}'.format(self.compiler.cxx))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sw4lite/package.py]: @ax3l @svenevs should this be `self.compiler.cxx` (the actual compiler) or `spack_cxx` (the compiler wrapper)?

File: var/spack/repos/builtin/packages/sw4lite/package.py [hunk 2]
```diff
@@ -52,6 +52,14 @@ def build_targets(self):
...
+            targets.append('MPIPATH= {0} '.format(self.spec['mpi'].prefix))
+            targets.append('gpuarch= {0}'.format(self.cuda_flags(cuda_arch)))
+            targets.append('MPIINC = {0}'.format(self.spec['mpi'].include))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/sw4lite/package.py]: Did you mean `prefix.include`? It might be best to use `headers.directories[0]` here.

Comment (rspavel) [reply var/spack/repos/builtin/packages/sw4lite/package.py]: Ooh, did not know that was a thing. That will make a lot of other things a lot easier. Fixed and thanks.

File: var/spack/repos/builtin/packages/sw4lite/package.py (hunk 1 above)
Comment (svenevs) [reply var/spack/repos/builtin/packages/sw4lite/package.py]: Hmm. I'm not really sure, I guess it feels more spack-like to make both the wrappers? I don't know of any other examples that need to do this though, I think everything else just relies on things being (injected) in the PATH?

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**